src/PacketaWidget/useTranslateCountry.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Getting country name from DOM and translating it to ISO code using AJAX.
+ *
+ * @package Packetery
+ */
+
+export const useTranslateCountry = (
+	adminAjaxUrl,
+	dynamicSettings,
+	setDynamicSettings,
+	loading,
+	setLoading,
+) => {
+
+	const getShippingCountryName = function () {
+		const inputElement = document.querySelector( '#shipping-country input' );
+		if ( inputElement ) {
+			return inputElement.value;
+		}
+		// keep initial country
+		return null;
+	};
+
+	const countryName = getShippingCountryName();
+	if ( countryName !== null ) {
+		if ( dynamicSettings === null || dynamicSettings.countryName === null ) {
+			setDynamicSettings( prevState => ( {
+				...prevState,
+				countryName,
+			} ) );
+		} else {
+			const previousCountryName = dynamicSettings.countryName;
+			if ( previousCountryName !== countryName ) {
+				if ( ! loading ) {
+					setLoading( true );
+					fetch( adminAjaxUrl, {
```

Comment:
Tento celý `useTranslateCountry` hook je de facto součástí renderu komponenty, ve které se použije - volá se při každém renderu. Není žádoucí dělat asynchronní operace přímo uvnitř renderu, tyhle věci je potřeba zabalit do `useEffect` hooku.

Reply:
Možná právě toho nežádoucího využívám, protože po zabalení do useEffectu to nereaguje na změnu země doručení.

Reply:
Pushni prosím tu variantu s useEffectem, mrknu na to. `useEffect` se volá při každém renderu, pokud nejsou uvedené žádné dependencies. Pokud jsou, tak při každé změně kterékoliv z nich.